### PR TITLE
feat(scale): add additional style properties

### PR DESCRIPTION
**Pull request by lorddevereux**
### Description of the feature or fix

See the issue which prompted this PR here [https://github.com/lvgl/lvgl/issues/6449](url)

New style parameters
- TRANSLATE_RADIAL
- RADIAL_OFFSET
- PAD_RADIAL

### Notes
- Documentation also updated to explain the new functions

### CODE REVIEW

File: src/widgets/scale/lv_scale.h [hunk 1]
```diff
@@ -54,6 +54,10 @@ typedef enum {
     LV_SCALE_MODE_LAST
 } lv_scale_mode_t;
 
+#define LV_SCALE_LABEL_ROTATE_MATCH_TICKS  0x100000
+#define LV_SCALE_LABEL_ROTATE_KEEP_UPRIGHT 0x80000
+#define LV_SCALE_ROTATION_ANGLE_MASK 0x7FFFF
```
Comment (kisvegabor) [inline src/widgets/scale/lv_scale.h]: Please add `LV_EXPORT_CONST_INT(LV_SCALE_LABEL_ROTATE_MATCH_TICKS) 0x100000` for all. It's required for the Micropython binding to see the defines.

**Review by kisvegabor — COMMENTED** [review]
Nice and clean implementation. Thank you! 

Could you also add 2 new examples (for linear and round)?

**Comment by kisvegabor** [comment]
Hey @lorddevereux,

Just a quick check if you will time to finalize this PR. If you are busy now we can take it over from here. 🙂

**Comment by lorddevereux** [comment]
Yes, sorry, been busy with work, will get to it this week!

**Comment by kisvegabor** [comment]
Great, thank you! 
(I closed it by accident)

**Comment by kisvegabor** [comment]
@lorddevereux Will you have time to finish it? We are so close :slightly_smiling_face:

**Comment by lorddevereux** [comment]
Sorry for the delay, I've added examples for linear and round - hopefully I did it correctly.

File: examples/widgets/scale/lv_example_scale_8.c [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    lv_obj_set_style_length(scale_line, 10, LV_PART_INDICATOR);
+    lv_scale_set_range(scale_line, 10, 40);
+
+    lv_scale_set_angle_range(scale_line, 270);
+    lv_scale_set_rotation(scale_line, 135);
+
+    lv_obj_t * needle_line = lv_line_create(scale_line);
+
+    lv_obj_set_style_line_width(needle_line, 6, LV_PART_MAIN);
+    lv_obj_set_style_line_rounded(needle_line, true, LV_PART_MAIN);
```
Comment (liamHowatt) [inline examples/widgets/scale/lv_example_scale_8.c]:
```suggestion
    lv_obj_t * needle_line = lv_line_create(scale_line);

    lv_obj_set_style_line_width(needle_line, 6, LV_PART_MAIN);
    lv_obj_set_style_line_rounded(needle_line, true, LV_PART_MAIN);
    static const lv_point_precise_t points[] = {{75, 75}, {50, 50}};
    lv_line_set_points(needle_line, points, 2);
```

Did you mean to set some points so the line is visible?

File: examples/widgets/scale/lv_example_scale_8.c
Comment (liamHowatt) [inline examples/widgets/scale/lv_example_scale_8.c:1]: Fyi, I noticed for both examples you set the same property twice a few times.

File: src/core/lv_obj_style_gen.c
Comment (liamHowatt) [inline src/core/lv_obj_style_gen.c:1]: This file is auto-generated. Please edit `style_api_gen.py` and then run `python3 scripts/style_api_gen.py` to automatically regenerate `lv_obj_style_gen.c` and `lv_obj_style_gen.h`.

**Review by liamHowatt — COMMENTED** [review]
Implementation looks great! Thanks for finding the time :slightly_smiling_face:

Maybe it's obvious but is there a way to set the radial offset of the labels? I don't think it's needed, just wondering.

File: src/widgets/scale/lv_scale.c [hunk 3]
```diff
@@ -636,11 +636,23 @@ static void scale_draw_label(lv_obj_t * obj, lv_event_t * event, lv_draw_label_d
...
+    int32_t translate_rotation = 0;
+
     if((LV_SCALE_MODE_VERTICAL_LEFT == scale->mode || LV_SCALE_MODE_VERTICAL_RIGHT == scale->mode)
        || (LV_SCALE_MODE_HORIZONTAL_BOTTOM == scale->mode || LV_SCALE_MODE_HORIZONTAL_TOP == scale->mode)) {
-        scale_get_label_coords(obj, label_dsc, tick_point_b, &label_coords);
+        lv_point_t label_origin;
+        label_origin.x = tick_point_b->x + translate_x;
+        label_origin.y = tick_point_b->y + translate_y;
+        scale_get_label_coords(obj, label_dsc, &label_origin, &label_coords);
+        label_rotation = (label_rotation & LV_SCALE_ROTATION_ANGLE_MASK) * 10U;
```
Comment (kisvegabor) [inline src/widgets/scale/lv_scale.c]: Please do not multiple by 10 to allow setting the angle with 0.1 degree precision and for consistency with other angle units.

File: examples/widgets/scale/lv_example_scale_8.c (hunk 2 above)
Comment (lorddevereux) [reply examples/widgets/scale/lv_example_scale_8.c]: Good spot I missed lv_scale_set_line_needle_value

Comment (lorddevereux) [reply examples/widgets/scale/lv_example_scale_8.c]: The radial offset of the labels can be adjusted using PAD_RADIAL, but this is missing from the docs. I will fix

**Review by liamHowatt — DISMISSED** [review]
Thanks!

**Review by kisvegabor — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Comment by kisvegabor** [comment]
Merged, thank you!

**Comment by clydebarrow** [comment]
Just noticed this code in master - but overloading the `transform_rotation` style property seems a bit kludgey to me - would not LV_SCALE_LABEL_ROTATE_MATCH_TICKS and `KEEP_UPRIGHT` be better as boolean object properties?

**Comment by liamHowatt** [comment]
It's the `transform_rotation` style property of the `LV_PART_INDICATOR` part. Other widgets also do this sort of creative "overloading" through dedicated `lv_part_t`s.

**Pull request merged**